### DISCUSSION TITLE
v3.0

### DISCUSSION THREAD

**Discussion by antonreshetov** [massCodeIO/massCode#185] [Announcements]
I am excited that v3.0 will be available soon, which now uses Codemirror instead of Ace. But this is not the main point, Codemirror in combination with `codemirror-textmate` allows the application to use `.tmLanguage` to highlight the syntax directly. This opens the door to almost [600](https://github.com/github/linguist/blob/master/vendor/README.md) existing grammars. Also, if the grammar is not correct, you just have to fix or replace the corresponding grammar and you no longer need to convert to the editor's specification. 

I put a lot of time and effort into making this tandem work. I also manually added [161](https://github.com/massCodeIO/massCode/tree/master/src/renderer/components/editor) grammars for 95%+ compatibility with the previous version.

On the downside, not all of the grammars used in Ace I found. 

Also now it is possible to use `.tmTheme` for themes, I have already added dark and light Material Theme. 

[v3.0.0-beta.1](https://github.com/massCodeIO/massCode/releases/tag/v3.0.0-beta.1). 

![image](https://user-images.githubusercontent.com/17480987/182623129-d9c8f4ba-23cb-4704-817a-60b1a8ea6439.png)

**Comment by HeliusHao** [comment]
Come on, look forward to the new version

**Comment by antonreshetov** [comment]
Just added the [v3.0.0-beta.1](https://github.com/massCodeIO/massCode/releases/tag/v3.0.0-beta.1)

**Reply by dvdmunckhof** [reply]
It works fine for me so far! ~~The only downside I've found so far, is that the Bash syntax highlighting is now gone. The whole file is just a single color.~~

**Edit**: it seems to depend on the script. Some Bash snippets are highlighted, some aren't. Same for Markdown, only URLs are highlighted.

**Reply by antonreshetov** [reply]
It's weird, but mine looks fine.
![image](https://user-images.githubusercontent.com/17480987/182821212-ba2d7bc2-24d5-45ee-8ed0-fa8c71868d7e.png)

**Reply by antonreshetov** [reply]
The v3 branch has a [database](https://github.com/massCodeIO/massCode/tree/v3/demo/db) for the test with all snippets and grammars. 

Anyone who can test it would be welcome 🙏

**Reply by dvdmunckhof** [reply]
I already edited my comment, but you were too fast with your reply :) It seems to depend on the script, example:
![image](https://user-images.githubusercontent.com/3891060/182822215-9f32298e-a8e4-418f-b1d9-8743459165f6.png)

I'm pretty sure this snippet used to have more highlighting in v2, I can't verify this, though. Same for Markdown, only URLs are highlighted.

**Reply by antonreshetov** [reply]
This is in v2
![image](https://user-images.githubusercontent.com/17480987/182823080-97d6046c-84ab-4b0e-97dd-2b2af574a64d.png)

**Reply by dvdmunckhof** [reply]
I did some comparisons with v2, the main difference is the base color is now light green instead of white, and dashes are not highlighted anymore (using Merbivore dark theme). Not a big deal, I preferred the old color scheme, though. Anyway, thanks for your work and fast replies!

v2:
![image](https://user-images.githubusercontent.com/3891060/182833270-456d966b-3298-4dbc-bbba-661e87d4df27.png)

v3:
![image](https://user-images.githubusercontent.com/3891060/182833337-759ac038-fdec-4584-af69-0af4c89d9d0d.png)

**Reply by antonreshetov** [reply]
Probably a little unaccustomed. By the way I checked in VS Code and the highlighting is the same as massCode )

![image](https://user-images.githubusercontent.com/17480987/182847154-4c7aa28c-a000-4d2a-aab6-2b5f4d7a7ef9.png)
![image](https://user-images.githubusercontent.com/17480987/182847218-45d619a7-f740-4220-b96d-10dcaeb486ae.png)

**Comment by alexn-s** [comment]
looks really nice. excited for the full release

**Comment by antonreshetov** [comment]
Just released [v3.0.0](https://github.com/massCodeIO/massCode/releases/tag/v3.0.0) 🎉
